src/operation-node/query-node.ts

```diff
@@ -43,6 +43,18 @@ export const QueryNode = freeze({
     )
   },
 
+  cloneWithEndModifier<T extends QueryNode>(
```

Comment:
let's add:

```ts
type HasEndModifiers = { endModifiers?: ReadonlyArray<OperationNode> }
```

at the top and...

```suggestion
  cloneWithEndModifier<T extends HasEndModifiers>(
```